core/vdbe/sorter.rs

```diff
@@ -41,17 +165,149 @@ impl Sorter {
                 &self.collations,
             )
         });
-        self.records.reverse();
-        self.next()
     }
-    pub fn next(&mut self) {
-        self.current = self.records.pop();
+}
+
+struct ChunkReader {
+    /// The reader for the chunk file.
+    reader: io::BufReader<File>,
+    /// The read buffer.
+    buffer: Vec<u8>,
+    /// The current length of the buffer.
+    buffer_len: usize,
+    /// The maximum number of values in a record in this chunk.
+    max_values_len: usize,
+    /// The current record.
+    current: Option<ImmutableRecord>,
+}
+
+impl ChunkReader {
+    fn new(fd: File, max_payload_size: usize, max_values_len: usize) -> Self {
+        Self {
+            reader: io::BufReader::new(fd),
```

Comment:
Provided that we already maintain a buffer in the `ChunkReader` itself, I'm 100% sure whether we need this additional buffering layer. Let me know what you think.